integration-test-groups/aws2/aws2-cw/src/test/java/org/apache/camel/quarkus/component/aws2/cw/it/Aws2CwTest.java

```diff
@@ -90,4 +97,148 @@ public void metric() {
+                Cw2Constants.METRIC_DIMENSION_VALUE, "odd"));
+
+        RestAssured.given()
+                .contentType(ContentType.JSON)
+                .body(data)
+                .post("/aws2-cw/send-metric-maps/" + namespace)
+                .then()
+                .statusCode(201);
+
+        Awaitility.await().pollInterval(1, TimeUnit.SECONDS).atMost(120, TimeUnit.SECONDS).until(
+                () -> {
+
+                    List<Datapoint> oddDatapoints = client.getMetricStatistics(
+                            GetMetricStatisticsRequest.builder()
+                                    .namespace(namespace)
+                                    .metricName(metricName)
+                                    .statistics(Statistic.SAMPLE_COUNT, Statistic.MINIMUM, Statistic.MAXIMUM)
+                                    .startTime(startTime)
+                                    .dimensions(Dimension.builder().name("type").value("odd").build())
+                                    .endTime(Instant.ofEpochMilli(System.currentTimeMillis() + 10000))
+                                    .period(30)
+                                    .build())
+                            .datapoints();
+
+                    List<Datapoint> evenDatapoints = client.getMetricStatistics(
+                            GetMetricStatisticsRequest.builder()
+                                    .namespace(namespace)
+                                    .metricName(metricName)
+                                    .statistics(Statistic.SAMPLE_COUNT, Statistic.AVERAGE)
+                                    .startTime(startTime)
+                                    .dimensions(Dimension.builder().name("type").value("even").build())
+                                    .endTime(Instant.ofEpochMilli(System.currentTimeMillis() + 10000))
+                                    .period(30)
+                                    .build())
+                            .datapoints();
+                    LOG.debug("Expecting some datapoints for metric " + namespace + "/" + metricName + " (type='odd'), got "
+                            + oddDatapoints);
+                    if (oddDatapoints.isEmpty()) {
+                        return false;
+                    }
+                    LOG.debug("Expecting some datapoints for metric " + namespace + "/" + metricName + " (type='even'), got "
+                            + evenDatapoints);
+                    if (evenDatapoints.isEmpty()) {
+                        return false;
+                    }
+
+                    Datapoint oddDp = oddDatapoints.get(0);
+                    Datapoint evenDp = evenDatapoints.get(0);
+
+                    if (!(oddDp.sampleCount().intValue() == 2
+                            && oddDp.minimum().intValue() == 2 * value + 1
+                            && oddDp.maximum().intValue() == 2 * value + 3)) {
+                        throw new RuntimeException("Unexpected odd datapoint " + oddDp
```

Comment:
Is there any specific reason to throw RuntimeException instead of using assert* methods ?

Reply:
I'm asserting the returned message in the test class. It doesn't feel good to call assert from runtime (not test) scope.

Reply:
Ok, think I get that. It's actually a `pollInterval()` block.